contracts/os/andromeda-kernel/src/execute.rs

```diff
@@ -713,6 +728,24 @@ pub fn unset_env(execute_ctx: ExecuteContext, variable: String) -> Result<Respon
         .add_attribute("variable", variable))
 }
 
+fn generate_unique_id(
+    deps: &mut DepsMut,
+    origin: &str,
+    block: &u64,
+    chain_id: &str,
+) -> Result<String, ContractError> {
+    // Create a unique string combining block, origin and a nonce
+    let nonce = NONCE.load(deps.storage)?;
+    let combined = format!("{}{}{}", block, origin, nonce);
+    // Calculate SHA256 hash
+    let mut hasher = Sha256::new();
```

Comment:
If we assume that the tuple of block, origin and nonce is unique is there a benefit to hashing here? It also restricts our ability to verify the ID is valid. Maybe a b64 encoding might look cleaner but otherwise we could just stringify the tuple in some form.

Reply:
https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.TransactionInfo.html

Interestingly, I wonder if this `env.transaction.index` field is unique per block?

Reply:
Looks like it
```rust
pub struct TransactionInfo {
    /// The position of this transaction in the block. The first
    /// transaction has index 0.
    ///
    /// This allows you to get a unique transaction indentifier in this chain
    /// using the pair (`env.block.height`, `env.transaction.index`).
    ///
    pub index: u32,
}
```

Reply:
Could we just append `chain_id` to `env.transaction.index` and use that as the ID?